### PR TITLE
Update UnitExtensions.cs

### PR DESCRIPTION
**Pull request by NightMoon032**
Add IsFacing
and fix the ManaPercent (for no mana champion)

### CODE REVIEW

File: Aimtec.SDK/Extensions/UnitExtensions.cs [hunk 1]
```diff
@@ -356,7 +356,37 @@ public static bool IsValidSpellTarget(this AttackableUnit unit)
         /// <param name="unit">The unit.</param>
         public static float ManaPercent(this Obj_AI_Base unit)
         {
-            return unit.Mana / unit.MaxMana * 100;
+            if (unit.Mana == 0)
+            {
+                return 100f;//For Garen and others no mana champion
```
Comment (LHS1140b) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: This doesn't make sense

Comment (NightMoon032) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: if someone code for garen and check the mana. in the old code it will always return 0

Comment (Insensitivity) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: why would you check the mana if you are making a garen script ?

Comment (NightMoon032) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: in some AIO (i think they will made the universal mana check system)

Comment (NightMoon032) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: unit.Mana  => need to changed to MaxMana...

Comment (Insensitivity) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: Their code practice is bad then, a class shouldn't implement an interface that it doesn't use.

Comment (LHS1140b) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: Yes, but that's in the script logic? You should not check for `manapercent()` on a champion that does not has any mana just use `ObjectManager.GetLocalPlayer().Mana` to check stuff like LeeSin energy and stuff?

Comment (NightMoon032) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: that not my problem, i just dont want to see some assembly is broken (for their problem)

Comment (LHS1140b) [inline Aimtec.SDK/Extensions/UnitExtensions.cs]: Ok, but please remove it in the PR.

**Pull request closed without merging**